lib/token.js

```diff
@@ -464,53 +514,89 @@ function getWithoutPrompt(sdk, oauthOptions, options) {
 function getWithPopup(sdk, oauthOptions, options) {
   var oauthParams = util.clone(oauthOptions) || {};
   util.extend(oauthParams, {
-    display: 'popup'
+    display: 'popup',
+    responseMode: 'okta_post_message'
   });
   return getToken(sdk, oauthParams, options);
 }
 
-function getWithRedirect(sdk, oauthOptions, options) {
-  oauthOptions = util.clone(oauthOptions) || {};
+function prepareOauthParams(sdk, oauthOptions) {
   var oauthParams = getDefaultOAuthParams(sdk, oauthOptions);
-  // If the user didn't specify a responseMode
-  if (!oauthOptions.responseMode) {
-    // And it's only an auth code request (responseType could be an array)
-    var respType = oauthParams.responseType;
-    if (respType.indexOf('code') !== -1 &&
-        (util.isString(respType) || (Array.isArray(respType) && respType.length === 1))) {
-        // Default the responseMode to query
-        util.extend(oauthParams, {
-          responseMode: 'query'
-        });
-    // Otherwise, default to fragment
-    } else {
-      util.extend(oauthParams, {
-        responseMode: 'fragment'
-      });
-    }
+
+  var responseType = oauthParams.responseType;
+  if (typeof responseType === 'string') {
+    responseType = [responseType];
+  }
+  if (responseType.includes('code')) {
```

Comment:
Left a comment in the README, but we need to remove this check. [Our documentation](https://developer.okta.com/authentication-guide/implementing-authentication/auth-code-pkce/#_2-using-the-authorization-code-flow-with-pkce) mentions that you **must** have this value when using this grant.

Instead - when we obtain values for the `grant_type` and `response_type`, we should assert that **if `grant_type` is `authorization_code` then `response_type` is `code`**. If it isn't, we can add that for them as the default value.

Therefore:
- `grant_type=authorization_code` && `response_type=token`: Fails
- `grant_type=authorization_code` && `response_type=id_token`: Fails
- `grant_type=authorization_code` && `response_type=code`: Succeeds
- `grant_type=authorization_code` && `response_type=undefined`: Succeeds as we populate this value for them.

Reply:
Yes this definitely makes sense. I am assuming that /token will always return both an accessToken and an idToken, since we are requiring `openid` scope and we do not support `offline_access`